internals/api/auth.go

```diff
@@ -0,0 +1,249 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/secrethub/secrethub-go/internals/api/uuid"
+)
+
+// AuthMethod options
+const (
+	AuthMethodAWSSTS = "aws-sts"
+)
+
+// SessionType options
+const (
+	SessionTypeHMAC SessionType = "hmac"
+)
+
+// Errors
+var (
+	ErrInvalidSessionType  = errAPI.Code("invalid_session_type").StatusError("invalid session type provided for authentication request", http.StatusBadRequest)
+	ErrInvalidPayload      = errAPI.Code("invalid_payload").StatusError("invalid payload provided for authentication request", http.StatusBadRequest)
+	ErrInvalidAuthMethod   = errAPI.Code("invalid_auth_method").StatusError("invalid auth method", http.StatusBadRequest)
+	ErrMissingField        = errAPI.Code("missing_field").StatusErrorPref("request is missing field %s", http.StatusBadRequest)
+	ErrSessionNotFound     = errAPI.Code("session_not_found").StatusError("session could not be found, it might have expired", http.StatusForbidden)
+	ErrSessionExpired      = errAPI.Code("session_expired").StatusError("session has expired", http.StatusForbidden)
+	ErrAuthFailed          = errAPI.Code("auth_failed").StatusError("authentication failed", http.StatusForbidden)
+	ErrCouldNotGetEndpoint = errAPI.Code("wrong_endpoint").StatusError("could not find the AWS endpoint", http.StatusBadRequest)
+	ErrAWSException        = errAPI.Code("aws_exception").StatusError("AWS returned an error", http.StatusFailedDependency)
```

Comment:
I understand your concerns and I agree this can be improved. This is returned if an AWS response we get on the server is unexpected. I don't think we want to expose any of this to the user, so the content of the error message should be vague (like an internal server error; though this one is not inernal). However, we could make the message a bit more descriptive like the error message for unexpected errors.

Reply:
Rephrased to `encountered an unexpected problem while verifying your identity on AWS. Please try again later.`. What do you think?